tt_metal/jit_build/CMakeLists.txt

```diff
@@ -8,3 +8,15 @@ set(JIT_BUILD_SRCS
 
 add_library(jit_build OBJECT ${JIT_BUILD_SRCS})
 target_link_libraries(jit_build PUBLIC common)
+
+execute_process(
+    COMMAND
+        git rev-parse --short=10 HEAD
+    RESULT_VARIABLE git_parse_result
+    OUTPUT_VARIABLE git_commit_hash_value
+    OUTPUT_STRIP_TRAILING_WHITESPACE
+)
+
+if(git_parse_result EQUAL "0")
+    target_compile_definitions(jit_build PUBLIC "-DGIT_COMMIT_HASH=\"${git_commit_hash_value}\"")
+endif()
```

Comment:
What happens if:

A) We are building from a tar.gz source blob and there is no .git directory? Will things still function?

B) Git is not installed?

I think we need to cover these two cases before this is merged.

Reply:
Yes, don't we have a general rule of not relying on source control to perform a build.

Looping in @afuller-TT as well

Reply:
if the command returns any error, the "GIT_COMMIT_HASH" will not be defined, and in that case we will create directories like how we do now, without any hash (built/.../), instead of "built/GIT_COMMIT_HASH/... "
I will test out the two cases you mentioned as well.

Reply:
Since it's a build parameter, how can we test this in a gtest?

Reply:
We do, but I'll leave that up to you guys.